### PR TITLE
Add coredump support

### PR DESCRIPTION
**Pull request by jakkra**
See `DEBUGGING_ISSUES.md` for how to use it.

Implements parts of https://github.com/jakkra/ZSWatch/issues/249

- Store coredump in RAM area when crashing
- At bootup, check if there is a coredump in RAM
- If so write it to external flash
- And a bunch of other nice things around it.

More coredump improvements to come

### CODE REVIEW

File: app/src/zsw_coredump.c [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+    const char *fileName;
+    struct crash_info_header header;
+
+    fileName = file;
+
+    if (strlen(file) > MAX_FILENAME_LEN) {
+        // Copy end of filename as it's more relevant than the path to the file.
+        fileName += strlen(file) - MAX_FILENAME_LEN;
+    }
+    strncpy(header.crash_file, fileName, MAX_FILENAME_LEN);
```
Comment (Kampi) [inline app/src/zsw_coredump.c:294]:
```
../src/zsw_coredump.c: In function 'assert_post_action':
../src/zsw_coredump.c:294:5: warning: '__builtin_strncpy' specified bound 32 equals destination size [-Wstringop-truncation]
  294 |     strncpy(header.crash_file, fileName, MAX_FILENAME_LEN);
      |     ^~~~~~~
```

File: .vscode/settings.json [hunk 2]
```diff
@@ -5,7 +5,9 @@
     ],
     "nrf-connect.boardRoots": [
         "${workspaceFolder}/app",
-        "${workspaceFolder}\\app"
+        "${workspaceFolder}\\app",
+        "${workspaceFolder}/zephyr/boards",
+        "${workspaceFolder}\\zephyr\\boards",
```
Comment (Kampi) [inline .vscode/settings.json:10]: Why double?

File: app/src/zsw_coredump.c (hunk 1 above)
Comment (jakkra) [reply app/src/zsw_coredump.c:294]: Why do I not get this warning?...

Comment (jakkra) [reply app/src/zsw_coredump.c:294]: Oh, I get it in release build, but not debug build Need to figure out why. Should show in debug also.

**Pull request merged**